### PR TITLE
add local_addr binding to connector service

### PR DESCRIPTION
**Pull request by fakeshadow**
## PR Type

Feature

## PR Checklist
Check your PR fulfills the following:

- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt

## Overview

re-export `tokio::net::TcpSocket` in `actix_rt::net` module.
Add local_addr binding to `actix_tls::connect` module.

### CODE REVIEW

File: actix-tls/src/connect/connector.rs [hunk 1]
```diff
@@ -148,11 +162,40 @@ impl<T: Address> Future for TcpConnectorResponse<T> {
...
+                }
+                IpAddr::V6(ip_addr) => {
+                    let socket = TcpSocket::new_v6()?;
+                    let addr = SocketAddr::V6(SocketAddrV6::new(ip_addr, 0, 0, 0));
+                    socket.bind(addr)?;
+                    socket
+                }
+            };
+
+            socket.set_reuseaddr(true)?;
```
Comment (robjtede) [inline actix-tls/src/connect/connector.rs:194]: can you drop a comment on this

**Review by robjtede — APPROVED** [review]
LGTM

File: actix-tls/src/connect/connector.rs (hunk 1 above)
Comment (fakeshadow) [reply actix-tls/src/connect/connector.rs:194]: This can be removed and default to false. A config for this can be add later if there is need.

**Comment by robjtede** [comment]
Weirdly, this cuased test failures on my local macOS. Fixed by doing `ifconfig lo0 alias 127.0.0.3`.

**Comment by robjtede** [comment]
And to my surprise, actix-tls tests are being skipped entirely on macOS CI.

**Pull request merged**